src/utils/c_utils/src/string_c_utils.c

```diff
@@ -466,9 +466,10 @@ size_t ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_
         return 0;
     }
 
-    if(numSrcCharsToCopy >= destByteLen)
+    // Ensure we don't overflow the destination buffer
```

Comment:
464: destByteLen better be at least 1 as resultant buffer always has at least a \0